viewing_party/party.py

```diff
@@ -1,23 +1,140 @@
+from statistics import mode
+
 # ------------- WAVE 1 --------------------
 
+
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        return None
+        
+    movie_dict = {
+        "title" : title,
+        "genre" : genre,
+        "rating" : rating
+    }
+
+    return movie_dict
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, movie):
+
+    for watchlist_movie in user_data["watchlist"]:
+        if watchlist_movie["title"] == movie:
+            user_data["watched"].append(watchlist_movie)
+            user_data["watchlist"].remove(watchlist_movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    total_of_ratings = 0
+    average_rating = 0
+
+    for movie in user_data["watched"]:
+        total_of_ratings += movie["rating"] 
+        average_rating = total_of_ratings / len(user_data["watched"])
+        
+    return average_rating
+
+
+def get_most_watched_genre(user_data):
+    all_watched_genres = []
+    if len(user_data["watched"]) == 0:
+        return None

```

Comment:
Let's move the guard clause all the way to the top. This way we aren't creating variables that might be immediately unused if the guard clause is executed.

```suggestion
    if len(user_data["watched"]) == 0:
        return None
    all_watched_genres = []
```